nisystemlink/clients/dataframe/_data_frame_client.py

```diff
@@ -275,14 +277,38 @@ def query_decimated_data(
         """Reads decimated rows of data from the table identified by its ID.
 
         Args:
-            id: Unique ID of a DataFrame table.
+            id: Unique ID of a data table.
             query: The filtering and decimation options to apply when reading data.
 
         Returns:
             The decimated table data.
 
         Raises:
-            ApiException: if unable to communicate with the Data Frame service
+            ApiException: if unable to communicate with the DataFrame Service
+                or provided an invalid argument.
+        """
+        ...
+
+    @response_handler()  # type: ignore
```

Comment:
I get a mypy error here that I haven't figured out how to resolve:
````
PS C:\dev\ni\nisystemlink-clients-python> poetry run poe types
Poe => mypy --config-file mypy.ini nisystemlink examples tests
nisystemlink\clients\dataframe\_data_frame_client.py:293: error: Untyped decorator makes function "iter_content_filelike_wrapper" untyped
nisystemlink\clients\dataframe\_data_frame_client.py:297: error: Untyped decorator makes function "export_table_data" untyped
```

Reply:
We nee to suppress type checking for this, because we don't have the ability to add typing for response_handler.

Reply:
Should that be a comment in the code? :)

Reply:
Done :)

Reply:
I added wrappers for the Uplink decorators here: https://github.com/ni/nisystemlink-clients-python/blob/master/nisystemlink/clients/core/_uplink/_methods.py
to add proper type annotations (and some default args). We could add `response_handler` there if we wanted to. It just pushes the `type: ignore` to a different module, though. My goal was to make the client implementation as smooth as possible. It's probably less important for this narrow use case.

Reply:
Actually, we definitely need to bolster the types somehow. The function signature and docstring are getting lost in translation for the export method:

https://user-images.githubusercontent.com/9257800/228989125-0e868847-2c4f-4dad-acf8-bd37e8dd8237.mov

Reply:
I think we just need to invert the `response_handler` usage:

```python
def _iter_content_filelike_wrapper(response: Response) -> IteratorFileLike:
    return IteratorFileLike(response.iter_content(chunk_size=4096))

@response_handler(_iter_content_filelike_wrapper) # type: ignore
@post("tables/{id}/export-data", args=[Path, Body])
def export_table_data(
    self, id: str, query: models.ExportTableDataRequest
) -> IteratorFileLike: ...
```

Now my intellisense is as expected:
<img width="641" alt="Screenshot 2023-03-30 at 6 57 09 PM" src="https://user-images.githubusercontent.com/9257800/228989993-3ec55081-48ee-441f-be2a-90005543ba2b.png">

Reply:
Thanks for the information! I added a response_handler wrapper to _methods and inverted the response handler usage.